server/application/application.go

```diff
@@ -1923,9 +1971,10 @@ func (s *Server) Rollback(ctx context.Context, rollbackReq *application.Applicat
 	if deploymentInfo == nil {
 		return nil, status.Errorf(codes.InvalidArgument, "application %s does not have deployment with id %v", a.QualifiedName(), rollbackReq.GetId())
 	}
-	if deploymentInfo.Source.IsZero() {
+	if deploymentInfo.Source.IsZero() && deploymentInfo.Sources.IsZero() {
```

Comment:
Can it be nil? It's just an empty slice, so checking `nil` or the len should be the same, and it doesn't panic because I'm not dereferencing it. You can check the behavior here: https://go.dev/play/p/UF45REm9OKE

Reply:
Thanks for the playground link. That helps understand it.